src/transformers/generation_tf_utils.py

```diff
@@ -2560,7 +2582,7 @@ def beam_search_cond_fn(
             # 3. is there still a beam that has not finished?
             still_open_beam = ~(tf.math.reduce_all(is_sent_finished) & early_stopping)
 
-            return not_max_length_yet & still_open_beam & improvement_still_possible
+            return not_max_length_yet & (still_open_beam | improvement_still_possible)
```

Comment:
This is the logic followed in the `done` function, in the previous implementation -- generation may continue if (and only if) `max_len` has not been reached, and then either because we haven't filled all the spots (when `early_stopping` is `True`) or because we can improve our beams.